src/main/java/org/folio/circulation/domain/policy/LoanPolicy.java

```diff
@@ -434,4 +444,75 @@ else if(isFixed(loansPolicy)) {
       return Optional.empty();
     }
   }
+
+  public HttpResult<Loan> recall(Loan loan) {
+    final JsonObject recalls = representation
+        .getJsonObject("requestManagement", new JsonObject())
+        .getJsonObject("recalls", new JsonObject());
+
+    final HttpResult<DateTime> minimumDueDateResult =
+        getDueDate("minimumGuaranteedLoanPeriod", recalls,
+            loan.getLoanDate(), null);
+
+    // Use a java.time Clock, which allows unit tests to use a fixed clock
+    final DateTime systemDate = new DateTime(
+        ZonedDateTime.now(
+            ClockManager.getClockManager().getClock()).toInstant().toEpochMilli(),
```

Comment:
I think we may want to consider adding our own abstraction around this to make the common uses simpler, but that is likely for a later day

Reply:
Yeah, this is ugly. My hope was that we would move to java.time sooner rather than later and then this code gets much simpler. However, that doesn't seem likely. I'll move this logic to that annoying `ClockManager` class. It's an easy, not risky, fix and will make the code a bit more palatable.